src/evict/evict_lru.c

```diff
@@ -1760,12 +1794,26 @@ __evict_walk_file(WT_SESSION_IMPL *session,
 			btree->evict_start_type =
 			    (btree->evict_start_type + 1) %
 			    WT_EVICT_WALK_START_NUM;
+			WT_STAT_CONN_INCR(
+			    session, cache_eviction_walks_gave_up);
+			WT_STAT_DATA_INCR(
+			    session, cache_eviction_walks_gave_up);
```

Comment:
I believe more granularity here is interesting. i.e: Whether it's because we didn't find any pages to queue, or if it's the ratio that failed - what that ratio was (somehow).